app/controllers/submit.js

```diff
@@ -899,15 +1019,70 @@ export default Ember.Controller.extend(Analytics, BasicsValidations, NodeActions
             const node = this.get('node');
             this.set('savingPreprint', true);
             this.toggleProperty('shareButtonDisabled');
-            model.set('isPublished', true);
+
+            let log = null;
+            if (this.get('moderationType')) {
+                log = this.get('store').createRecord('reviewLog', {
+                   action: 'submit',
+                   reviewable: this.get('model')
+                });
+            } else {
+                model.set('isPublished', true);
+            }
             node.set('public', true);
 
+            if (log) {
+                return log.save()
```

Comment:
@chrisseto, turns out the order does matter since the log request changes the state of the preprint. Duplicating this much code feels weird though. Do you know of a better way?

Reply:
@chrisseto, @aaxelb, changes made 💸